contributing/giving-constructive-criticism.rst

```diff
@@ -0,0 +1,155 @@
+is a great way to get started with contributing to the Symfony community.
+Anyone can do it! But before you give a comment, take a step back and think,
+is what you are about to say actually what you intend?
+
+Communicating over the Internet with nothing but text can pose a
+big challenge, especially if you remember that the Symfony community
+is world-wide and is composed of a wide variety of people with differing
+ideas and opinions.
+
+Not everyone speaks English or is able to use a keyboard, some might
+have Dyslexia or similar conditions that affect their writing.
+
+Not to mention that some might have a bad experience from previous
+contributions (to other projects).
+
+You're not alone in this. This guide will try to help you write
+constructive, respectful and helpful reviews and replies.
+
+.. tip::
+
+    This guide is not about lecturing you to "conform" or give-up
+    your ideas and opinions but helping you to better communicate,
+    prevent possible confusion, and keeping the Symfony community a
+    welcoming place for everyone. **You are free to disagree with
+    someone's opinions, just don't be disrespectful.**
+
+First of, accept that many programming decisions are opinions.
+Discuss trade offs, which you prefer, and reach a resolution quickly.
+It's not about being right or wrong, but using what works.
+
+Tone of voice
+-------------
+
+We don't expect you to be completely formal, or to even write error-free
+English. Just remember one thing: Don't swear, and be respectful to others.
+
+Don't reply in anger or with an aggressive tone. You're angry, we understand
+that, but swearing/cursing and name calling doesn't really encourage anyone to
+help you. Take a deep breath, count to 10 and try to *clearly* explain what problems
+you encounter.
+
+Gender pronouns
+---------------
+
+While not "formally" required it's better to use gender-neutral pronouns.
+Unless someone "indicated" their pronouns, use "they", "them" instead of
+"he", "she", "his", "hers", "his/hers", "he/she", etc.
+
+And instead of saying "guys" it's better to use "folks", or Symfonions
+if you like that.
+
+Giving positive feedback
+------------------------
+
+While reviewing issues and pull request you may run into some suggestions
+(including patches) that don't reflect your ideas, are not good, or downright wrong.
+
+Now, instead of writing a lengthy comment why you think they are wrong,
+consider the amount of work and time the author has spent on their idea
+and how you're response would make them feel;
+
+* Are you making a point only because you believe they are wrong?
+
+* Did you correctly understand their intention? or are you making assumptions?
+
+Maybe you can ask for clarification? suggest an alternative? or
+provide a simple explanation *why* disagree with their proposal.
+
+Even if something *is* really wrong or "a bad idea", stay respectful and
+don't get into endless you-are-wrong discussions or "flame wars".
```

Comment:
I think we should overemphasize that reviewing things **requires you to give technical reasons and facts, not opinions** (except in some edge cases):

* NO --> "I don't like how you wrote this code"
* YES --> "This code doesn't comply with Symfony's CS rules"
* EDGE CASE (it can be an opinion and/or a fact) --> "I think the code is less readable now"

Example of valid reasons to reject:

* We tried that in the past (link to the relevant PR) but we needed to revert it for XXX reason.
* That change would introduce too many merge conflicts when merging up Symfony branches. In the past we've always rejected changes like this.
* I profiled this change and it hurts performance significantly *(if you don't profile, it's an opinion, so we can ignore)*
* Code doesn't match Symfony's CS rules *(e.g. use `array()` instead of `[]`)*
* We only provide integration with very popular projects *(e.g. we integrate Bootstrap but not your own CSS framework)*
* This would require adding lots of code and making lots of changes for a feature that doesn't look so important. That could hurt maintaining in the future.
* etc.